### PR TITLE
ReorgYolo reference implementation

### PR DESCRIPTION
**Pull request by mitruska**
[ReorgYolo](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/detection/ReorgYolo_1.md) reference implementation

### CODE REVIEW

**Comment by jane-intel** [comment]
ReorgYolo is a special case of ExtractImagePatches operation. Please see how we [convert EIP to ReorgYolo](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/transformations/src/transformations/convert_extract_image_patches_to_reorg_yolo.cpp)

I find it a bit more efficient to have reference for common operation -- EIP, and then use it for both ReogYolo and EIP.
@mitruska Does it sound sane to you?

**Comment by mitruska** [comment]
> I find it a bit more efficient to have reference for common operation -- EIP, and then use it for both ReogYolo and EIP.
> @mitruska Does it sound sane to you?

Thank you for the comment @jane-intel. I think it sounds reasonable, but we have ReorgYolo and ExtractImagePatches separately in the specification and plugins so from the validation and educational point of view, it would be better to have dedicated reference implementation for each of them rather than one for both.

File: ngraph/test/runtime/interpreter/int_executable.hpp [hunk 1]
```diff
@@ -892,6 +893,16 @@ class INTERPRETER_BACKEND_API ngraph::runtime::interpreter::INTExecutable : publ
                                              pbox->get_attrs());
             break;
         }
+        case OP_TYPEID::ReorgYolo_v0:
```
Comment (mbencer) [inline ngraph/test/runtime/interpreter/int_executable.hpp:896]: As I understand all method which have evaluate are calculated during Interpreter execution (look here: https://github.com/mbencer/openvino/blob/926be8356887567e1f58a7e55bc34a01cc660ed1/ngraph/test/runtime/interpreter/int_executable.cpp#L205). Why is it called directly?

Comment (mitruska) [reply ngraph/test/runtime/interpreter/int_executable.hpp:896]: It was recommended to not create evaluate method by default, so ReorgYolo does't have `evaluate`.

Comment (jdanieck) [reply ngraph/test/runtime/interpreter/int_executable.hpp:896]: This op won't be used in Constant Folding, hence it should be called via int_executable (not via evaluate()) to not unnecessarily increase ngraph binary size.

Comment (mbencer) [reply ngraph/test/runtime/interpreter/int_executable.hpp:896]: right, I miss it ;)

**Review by postrational — APPROVED** [review]

**Review by jane-intel — APPROVED** [review]

**Comment by jane-intel** [comment]
@mitruska, would you like to update specification in this PR, or in a separate PR?

**Comment by mitruska** [comment]
@jane-intel I will create a separate PR for the ReorgYolo specification review.

File: ngraph/core/src/op/reorg_yolo.cpp [hunk 2]
```diff
@@ -15,6 +15,8 @@
 //*****************************************************************************
 
 #include "ngraph/op/reorg_yolo.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
```
Comment (jdanieck) [inline ngraph/core/src/op/reorg_yolo.cpp:18]: Looks like these includes are not needed anymore.

File: ngraph/test/backend/reorg_yolo.in.cpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    try
+    {
+        auto test_case = test::TestCase<TestEngine>(fun);
+        test_case.add_input<float>(inputs);
+        test_case.add_expected_output<float>(expected_shape, fake_output);
+        test_case.run();
+
+        // Should have thrown, so fail if it didn't
+        FAIL() << "Incompatible stride was not detected.";
```
Comment (jdanieck) [inline ngraph/test/backend/reorg_yolo.in.cpp:128]: Did you consider using [exception assertions](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#exception-assertions) ?

Comment (mitruska) [reply ngraph/test/backend/reorg_yolo.in.cpp:128]: I think `ASSERT_THROW(test_case.run(), ngraph_error)` is too general `ASSERT_ANY_THROW()` even more.
The test will pass even the error is not about stride,

**Review by jdanieck — APPROVED** [review]

**Review by ggalieroc — APPROVED** [review]

**Review by iefode — APPROVED** [review]
IE Test part LGTM

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**